apps/src/lib/levelbuilder/data-docs-editor/DataDocFormEditor.jsx

```diff
@@ -0,0 +1,93 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import RailsAuthenticityToken from '@cdo/apps/lib/util/RailsAuthenticityToken';
+import HelpTip from '@cdo/apps/lib/ui/HelpTip';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import {navigateToHref} from '@cdo/apps/utils';
+
+const DataDocFormEditor = props => {
+  const {dataDocKey, originalDataDocName, originalDataDocContent} = props;
+  const [dataDocContent, setDataDocContent] = useState(originalDataDocContent);
+  const [dataDocName, setDataDocName] = useState(originalDataDocName);
+
+  const save = () => {
+    $.ajax({
+      url: `/data_docs/${dataDocKey}`,
+      method: 'PUT',
+      data: {
+        name: dataDocName,
+        content: dataDocContent
+      }
+    }).done(() => {
+      navigateToHref(`/data_docs/${dataDocKey}`);
+    });
+  };
+
+  return (
+    <div>
+      <RailsAuthenticityToken />
+      <h1>Edit Data Doc</h1>
+      <h2>
+        This feature is in progress. It is not ready for use on Levelbuilder
+        yet.
+      </h2>
+      <label style={styles.label}>
+        Slug
+        <input
+          className="input"
+          name="key"
+          style={styles.input}
+          value={dataDocKey}
+          disabled
+        />
+        <HelpTip style={styles.helptip}>
+          <p>
+            The data doc slug is used in URLs and cannot be updated once set. A
+            slug can only contain lowercase letters, numbers, and dashes, and
+            'new' and 'edit' are reserved.
+          </p>
+        </HelpTip>
+      </label>
+      <label style={styles.label}>
+        Name
+        <input
+          className="input"
+          name="name"
+          style={styles.input}
+          value={dataDocName}
+          onChange={e => setDataDocName(e.target.value)}
+        />
+      </label>
+      <TextareaWithMarkdownPreview
+        name="content"
+        label="Content"
+        handleMarkdownChange={e => setDataDocContent(e.target.value)}
+        markdown={dataDocContent || ''}
+      />
+      <br />
+      <button className="btn btn-primary" type="submit" onClick={save}>
```

Comment:
It would be great if we could use the SaveBar component here and wire this up to support both Save and Close  as well as Save and Keep Editing

Reply:
SaveBar attaches to the bottom of the screen instead of being at the end of the form so I'll include its functionality but just in the form of an additional Save and Keep Editing button

Reply:
Scratch that! Going to switch to SaveBar